src/GovUk.Education.ExploreEducationStatistics.Admin/Services/ReleaseService.cs

```diff
@@ -322,18 +324,31 @@ public class ReleaseService : IReleaseService
                     {
                         await _publishingService.ReleaseChanged(releaseId, request.PublishMethod == PublishMethod.Immediate);
                     }
-                    
+
                     _context.Update(release);
                     await _context.SaveChangesAsync();
 
                     return await GetReleaseForIdAsync(releaseId);
                 });
         }
 
+        private async Task<Either<ActionResult, Unit>> ValidateSelectedPublication(Release release, Guid publicationId)
+        {
+            if (release.PublicationId == publicationId)
+            {
+                return Unit.Instance;
+            }
+
+            return await _persistenceHelper.CheckEntityExists<Publication>(publicationId)
+                .OnFailureFailWith(() => ValidationActionResult(PublicationDoesNotExist))
+                .OnSuccess(_userService.CheckCanCreateReleaseForPublication)
```

Comment:
Unfortunately don't think this will help in this specific case as `CheckCanCreateReleaseForPublication` returns the publication it receives if it's successful. I'll change this method anyway to just return the publication so we don't need to bother re-mapping it to Unit.

In the case of `CheckCanAccessSystem`, these are resource-less checks so they don't return anything so Unit does make sense in those cases. I've updated them to return Unit instead of boolean.